server/controllers/data/lib/summaries/openlibrary.js

```diff
@@ -1,31 +1,68 @@
 import { cache_ } from '#lib/cache'
+import { normalizeIsbn } from '#lib/isbn/isbn'
 import { requests_ } from '#lib/requests'
+import { requireJson } from '#lib/utils/json'
+
+const wmCodeByIso6392Code = requireJson('wikidata-lang/mappings/wm_code_by_iso_639_2_code.json')
 
 const timeout = 10 * 1000
 
-export default async ({ id, refresh }) => {
-  const lastLetter = id.slice(-1)[0]
-  const section = openLibrarySectionByLetter[lastLetter]
-  const link = `https://openlibrary.org/${section}/${id}`
-  const url = `${link}.json`
-  const property = 'wdt:P648'
-  const text = await cache_.get({
-    key: `summary:${property}:${id}`,
+export async function getOpenLibrarySummary ({ claims, refresh }) {
+  const id = claims['wdt:P648']?.[0]
+  const isbn13h = claims['wdt:P212']?.[0]
+  let link, url, property, key, cacheKey, claim
+  if (id) {
+    const lastLetter = id.slice(-1)[0]
+    const section = openLibrarySectionByLetter[lastLetter]
+    link = `https://openlibrary.org/${section}/${id}`
+    url = `${link}.json`
+    property = 'wdt:P648'
+    key = property
+    cacheKey = `summary:${key}:${id}`
+    claim = { id, property }
+  } else if (isbn13h) {
+    const isbn13 = normalizeIsbn(isbn13h)
+    link = `https://openlibrary.org/isbn/${isbn13}`
+    url = `${link}.json`
+    property = 'wdt:P212'
+    key = `${property}:openlibrary`
+    cacheKey = `summary:${key}:${isbn13}`
+    claim = { id: isbn13h, property }
+  } else {
+    return
+  }
+  const res = await cache_.get({
+    key: cacheKey,
     refresh,
     fn: async () => {
-      const { bio, description } = await requests_.get(url, { timeout })
-      const text = bio || description
-      if (!text) return
-      if (text.value) return text.value
-      else if (typeof text === 'string') return text
+      try {
+        const { bio, description, languages } = await requests_.get(url, { timeout })
+        const text = bio || description
+        if (!text) return
```

Comment:
OpenLibrary is kind enough to accept ISBN-13 in place of ISBN-10, see for your first example https://openlibrary.org/isbn/9780517588475. But your examples raises an other issue: the edition itself doesn't have a description, but its work does, and it would be nice to get that instead of nothing.